### PR TITLE
Add datasource.type to widgets, and support for Cloud SQL

### PR DESCRIPTION
**Pull request by jmuharsky**

### CODE REVIEW

File: client/components/widget/data_viz/gviz/gviz-directive_test.js [hunk 1]
```diff
@@ -84,7 +84,10 @@ describe('gvizDirective', function() {
                 /.*pk-cond-(\S+)/, '$1');
 
             var result = {};
-            result.pass = (parentState == expected);
+            var classNames = parentState.split(' ');
```
Comment (klausw) [inline client/components/widget/data_viz/gviz/gviz-directive_test.js]: Did something get out of sync, or is this a side effect of splitting branches? This shouldn't be part of this PR, merges should IMHO be separate.

**Comment by klausw** [comment]
LGTM as far as the change is concerned, but the PR description is a bit confusing since it includes two unrelated fixup patches.

**Comment by jmuharsky** [comment]
The patches aren't actually "in" this branch; they were merged from master before being checked in, and I'm having difficulty getting them to rebase/vanish.  As the result is as expected, I'm not inclined to push further.
